### PR TITLE
Support unknowns in pedigree

### PR DESCRIPTION
**Pull request by bpblanken**

### CODE REVIEW

File: v03_pipeline/lib/misc/family_loading_failures_test.py [hunk 1]
```diff
@@ -202,7 +202,6 @@ def test_get_families_failed_sex_check(self):
             [
                 [
                     'Sample ROS_006_18Y03226_D1 has pedigree sex F but imputed sex M',
-                    'Sample ROS_006_18Y03227_D1 has pedigree sex M but imputed sex F',
```
Comment (bpblanken) [inline v03_pipeline/lib/misc/family_loading_failures_test.py:205]: the logic behind this change... `ROS_006_18Y03227_D1` is now denoted as `U` in the test pedigree, so the family fails for only the `'Sample ROS_006_18Y03226_D1 has pedigree sex F but imputed sex M'` reason.

File: v03_pipeline/lib/misc/family_loading_failures.py [hunk 2]
```diff
@@ -175,7 +175,10 @@ def get_families_failed_sex_check(
     failed_families = defaultdict(list)
     for family in families:
         for sample_id in family.samples:
-            if family.samples[sample_id].sex != sex_check_lookup[sample_id]:
+            if family.samples[sample_id].sex not in {
+                sex_check_lookup[sample_id],
+                Sex.UNKNOWN,
+            }:  # NB: Unknown samples in pedigree are excluded from sex check.
```
[the remark this answers is not in the document]
Comment (matren395) [reply v03_pipeline/lib/misc/family_loading_failures.py:181]: remind me: so the sex_check_lookup is what our DSP/GP friends are delivering, right. So if a given sample is 'Unknown' in Seqr, it doesn't matter what it's returned as in our DSP/GP friends' metrics.tsv file thing. 

Would we want to try and ensure that it's also returned as 'U' by them? But once again, I don't know how that would look like returned 

This does still fail of the family.samples[sample_id].sex is M/F and sex_check_lookup[sample_id] is F/M, or vice-versa.

[the remark this answers is not in the document]
Comment (bpblanken) [reply v03_pipeline/lib/misc/family_loading_failures.py:181]: I think my thinking on this was that a `U` in the pedigree is quite distinct from a `U` in the imputed sex file and that they aren't actually comparable.  The former is a project management issue whereas the latter is likely a statistical issue within DRAGEN.  

I don't actually know how a `U` would be delivered in the imputed sex file, so we were, on purpose,  [_extremely strict_](https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/misc/io.py#L166-L174) with the import of that column.

[the remark this answers is not in the document]
Comment (matren395) [reply v03_pipeline/lib/misc/family_loading_failures.py:181]: Heard! In that case, lgtm

**Review by matren395 — APPROVED** [review]
😮

**Pull request merged**